src/invoice_utils/engine/_engine.py

```diff
@@ -33,6 +25,17 @@ def __init_invoice(self):
             "totals": {"price": 0, "total": 0, "extra": {}},
         }
 
+    @staticmethod
+    def __set_default_rules():
```

Comment:
This class accepts the `rules` directly. This `@staticmethod` doesn't belong here anymore because i/o results in the `rules` being passed to instances of this class. 
One other concern is that this method  causes the `InvoicingEngine` initializer to fail. The calling code fails implicitly. Since explicit is always better than implicit, failing explicitly while determining the rule template file to use is preferred.